examples/threading/threading.c

```diff
@@ -28,6 +63,27 @@ bool start_thread_obtaining_mutex(pthread_t *thread, pthread_mutex_t *mutex,int
      * 
      * See implementation details in threading.h file comment block
      */
-    return false;
+     
+    struct thread_data *params = malloc(sizeof(struct thread_data));
+
+    params->thread = thread;
+    params->wait_obtain_ms = wait_to_obtain_ms;
+    params->wait_release_ms = wait_to_release_ms;
+ 	params->lock = mutex;
+    params->thread_complete_success = false;
+
+    int ret = pthread_create(thread, NULL, &threadfunc, (void *)params);
+    if (ret != 0)
+    {
+        ERROR_LOG("pthread_create failed");
```

Comment:
@MuktaDarekar who frees `params` in this case?

Reply:
done.